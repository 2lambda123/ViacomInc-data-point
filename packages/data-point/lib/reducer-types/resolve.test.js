/* eslint-env jest */
'use strict'

const Factory = require('./factory')
const Resolve = require('./resolve')

const reducers = require('../../test/utils/reducers')

const fixtureStore = require('../../test/utils/fixture-store')
const testData = require('../../test/data.json')

const AccumulatorFactory = require('../accumulator/factory')

let manager

beforeAll(() => {
  manager = fixtureStore.create()
})

describe('reducer#resolve', () => {
  test('It should work for valid input', () => {
    const accumulator = AccumulatorFactory.create({
      value: testData.a.b.c
    })

    const reducer = Factory.create(reducers.addCollectionValues())
    return Resolve.resolve(manager, accumulator, reducer).then(result => {
      expect(result.value).toEqual(6)
    })
  })

  test('It should throw error for invalid input', () => {
    const accumulator = AccumulatorFactory.create({
      value: testData.a.b.c
    })

    const reducer = { type: 'INVALID TYPE' }
    expect(() => {
      Resolve.resolve(manager, accumulator, reducer)
    }).toThrowErrorMatchingSnapshot()
  })
})
