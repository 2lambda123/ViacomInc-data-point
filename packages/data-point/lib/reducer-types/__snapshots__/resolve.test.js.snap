// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer#resolve It should throw error for invalid input 1`] = `[Error: Reducer type 'INVALID TYPE' was not recognized]`;
